documentation/src/main/asciidoc/configuration.asciidoc

```diff
@@ -894,7 +894,51 @@ hibernate.search.[default|<indexname>].reader.strategy = my.corp.myapp.CustomRea
 
 where my.corp.myapp.CustomReaderProvider is the custom strategy implementation.
 
+=== Serialization
 
+When using clustering features, Hibernate Search needs to find an implementation of the
+`SerializationProvider` service on the classpath.
+
+An implementation of the service for link:https://avro.apache.org[Apache Avro] can be found
+using the following GAV coordinates:
+
+`org.hibernate:hibernate-search-serialization-avro:{hibernateSearchVersion}`
+
+You can add the coordinates to your pom file or download all the required dependecies and add them
+to your classpath. Hibernate Search will find the service implementation without any additional
+configuration.
+
+Alternatively, you can create a custom service implementation:
+
+[[example-serialization-provider]]
+.Serialization strategy definition
+====
+[source, JAVA]
+----
+import org.hibernate.search.indexes.serialization.spi.Deserializer;
+import org.hibernate.search.indexes.serialization.spi.SerializationProvider;
+import org.hibernate.search.indexes.serialization.spi.Serializer;
+
+public class ExampleOfSerializationProvider implements SerializationProvider {
+
+   @Override
+    public Serializer getSerializer() {
+        Serializer serializer = ...
+        return serializer;
+    }
+
+    @Override
+    public Deserializer getDeserializer() {
+        Deserializer deserializer = ...
+        return deserializer;
+    }
+}
+----
+====
+
+Once you have an implementation of the service you need to register it.
+If you are not familiar with this step, you will find more details about it in the
```

Comment:
Isn't SerializationProvider a service?
Do you mean that if I add a custom implementation I don't have to add the file under META-INF?